testsuite/smoke-test/src/key_manager.rs

```diff
@@ -3,68 +3,66 @@
-    key_manager_config.json_rpc_endpoint =
-        format!("http://127.0.0.1:{}", node_config.json_rpc.address.port());
-    key_manager_config.rotation_period_secs = 10;
-    key_manager_config.sleep_period_secs = 1000; // Large sleep period to force a single rotation
+    // Fetch the first node config in the swarm
+    let node_config_path = env.validator_swarm.config.config_files.get(0).unwrap();
+    let node_config = NodeConfig::load(&node_config_path).unwrap();
 
-    // Load validator's on disk storage and update key manager secure backend in config
-    let storage: Storage = if let Identity::FromStorage(storage_identity) =
-        &node_config.validator_network.as_ref().unwrap().identity
-    {
-        let storage_backend = storage_identity.backend.clone();
-        key_manager_config.secure_backend = storage_backend.clone();
-        (&storage_backend).try_into().unwrap()
-    } else {
-        panic!("Couldn't load identity from storage");
-    };
-
-    // Save the key manager config to disk
-    let key_manager_config_path = config_path.with_file_name("key_manager.yaml");
-    key_manager_config.save(&key_manager_config_path).unwrap();
+    // Load validator's on disk storage
+    let secure_backend = load_backend_storage(&env.validator_swarm, 0);
+    let storage: Storage = (&secure_backend).try_into().unwrap();
 
     // Create a json-rpc connection to the blockchain and verify storage matches the on-chain state.
-    let libra_interface = get_json_rpc_libra_interface(&env.validator_swarm, 0);
+    let json_rpc_endpoint = format!(
+        "http://127.0.0.1:{}",
+        env.validator_swarm.get_client_port(0)
+    );
+    let libra_interface = JsonRpcLibraInterface::new(json_rpc_endpoint.clone());
     let account = node_config.validator_network.unwrap().peer_id();
     let current_consensus = storage.get_public_key(CONSENSUS_KEY).unwrap().public_key;
     let validator_info = libra_interface.retrieve_validator_info(account).unwrap();
     assert_eq!(&current_consensus, validator_info.consensus_public_key());
 
-    // Spawn the key manager and sleep until a rotation occurs.
-    let mut command = Command::new(workspace_builder::get_bin(KEY_MANAGER_BIN));
-    command
-        .current_dir(workspace_builder::workspace_root())
-        .arg(key_manager_config_path);
+    // Create the key manager
+    let key_manager_storage: Storage = (&secure_backend).try_into().unwrap();
+    let mut key_manager = KeyManager::new(
+        JsonRpcLibraInterface::new(json_rpc_endpoint),
+        key_manager_storage,
+        RealTimeService::new(),
+        1,
+        1000, // Large sleep period to force a single rotation
+        1000,
+        ChainId::test(),
+    );
 
-    let mut key_manager = command.spawn().unwrap();
-    sleep(Duration::from_secs(20));
+    // Spawn the key manager and execute a rotation
+    sleep(Duration::from_secs(10)); // Wait a few seconds for the key to become stale
```

Comment:
this feels awkward, can we change the timeouts in key manager to be aggressive and then here we just poll every second for up to 20 seconds that the rotation occurred instead of this wait 10 and then wait 15 and then check?

Reply:
Sure, I've updated it to check in a loop for up to 60 seconds (6 seconds sleep period, 10 periods) 😄